snuba/query/snql/parser.py

```diff
@@ -632,11 +646,65 @@ def parse(exp: Expression) -> Expression:
     query.transform_expressions(parse)
 
 
+ARRAY_JOIN_MATCH = FunctionCallMatch(
+    StringMatch("arrayJoinExplode"),
+    (
+        Param("column", ColumnMatch(AnyOptionalString(), AnyMatch(str))),
+        Param("op", Or([LiteralMatch(StringMatch(op)) for op in OPERATOR_TO_FUNCTION])),
+        Param("value", AnyExpression()),
+    ),
+)
+
+
+def _array_join_transformation(query: LogicalQuery) -> None:
```

Comment:
Any reason for not defining this as a dataset custom function? Is it in order to avoid adding it manually everywhere or is there a reason why this has to happen here ?

Reply:
I put this here because custom functions are properties of entities (the processors are instantiated by the entity) and this is not specific to an entity, it is specific to SnQL.

Reply:
ok to me.